dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java

```diff
@@ -81,22 +85,29 @@ public void encode(BeamSQLRow value, OutputStream outStream,
         case FLOAT:
           doubleCoder.encode((double) value.getFloat(idx), outStream, context.nested());
           break;
+        case DECIMAL:
+          bigDecimalCoder.encode(value.getBigDecimal(idx), outStream, context.nested());
+          break;
         case BIGINT:
           longCoder.encode(value.getLong(idx), outStream, context.nested());
           break;
         case VARCHAR:
           stringCoder.encode(value.getString(idx), outStream, context.nested());
           break;
+        case TIME:
+          longCoder.encode(value.getGregorianCalendar(idx).getTime().getTime(),
+              outStream, context.nested());
+          break;
         case TIMESTAMP:
-          longCoder.encode(value.getDate(idx).getTime(), outStream, context);
+          longCoder.encode(value.getDate(idx).getTime(), outStream, context.nested());
           break;
 
         default:
           throw new UnsupportedDataTypeException(value.getDataType().getFieldsType().get(idx));
       }
     }
 
-    instantCoder.encode(value.getWindowStart(), outStream, context.nested());
+    instantCoder.encode(value.getWindowStart(), outStream, context);
```

Comment:
I would follow the pattern to have only the last field as `context`, and `context().nested()` with others.